### PR TITLE
fix: use basename for document title (#1630)

### LINKED ISSUE #1630: Name of marimo notebook tab title in browser should just be the name of the file (not include filepath)
### Describe the bug

I have 10+ marimo notebooks open, and I can't tell which one is which because the tab title starts with the name of the directory (since they are nested 2 directories deep from where I ran `marimo edit`)

Screenshot 2024-06-17 at 2 08 35 PM

### Environment

0.6.19

### Code to reproduce

_No response_

### PR DESCRIPTION
**Pull request by ericjanto**
## 📝 Summary

This PR fixes #1630 by displaying the basename only instead of the entire path in the tab.

## 🔍 Description of Changes

I introduced a new atom `useFiletitle()` which simply extracts the string after the last `/` character in the path (path.basename not available in the client). It depends on the full filename so reacts to changes in the filename accordingly.

## 📋 Checklist

- [x] I have read the [contributor guidelines](../CONTRIBUTING.md).
- [x] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://discord.gg/JE7nhX6mD8), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [x] I have added tests for the changes made.
- [x] I have run the code and verified that it works as expected.

## 📜 Reviewers

@akshayka OR @mscolnick

### CODE REVIEW

File: frontend/src/core/saving/filename.ts [hunk 1]
```diff
@@ -5,6 +5,15 @@ import { getFilenameFromDOM } from "../dom/htmlUtils";
 
 const filenameAtom = atom<string | null>(getFilenameFromDOM());
 
+const filetitleAtom = atom((get) => {
```
Comment (mscolnick) [inline frontend/src/core/saving/filename.ts:8]: i dont think we need to use a jotai atom for this. this is usually best if you need to hold expensive state or react to updates from it. 

instead we can inline this logic

File: frontend/src/core/edit-app.tsx [hunk 2]
```diff
@@ -119,7 +120,8 @@ export const EditApp: React.FC<AppProps> = ({ userConfig, appConfig }) => {
       .then(() => {
         setFilename(name);
         // Set document title: app_title takes precedence, then filename, then default
-        document.title = appConfig.app_title || name || "Untitled Notebook";
+        document.title =
+          appConfig.app_title || filetitle || "Untitled Notebook";
```
Comment (mscolnick) [inline frontend/src/core/edit-app.tsx:124]: you could do `Paths.basename` - this should handle windows too:

`appConfig.app_title || Paths.basename(name) || "Untitled Notebook";`

File: frontend/src/core/edit-app.tsx [hunk 3]
```diff
@@ -131,8 +133,8 @@ export const EditApp: React.FC<AppProps> = ({ userConfig, appConfig }) => {
   // Update document title whenever filename or app_title changes
   useEffect(() => {
     // Set document title: app_title takes precedence, then filename, then default
-    document.title = appConfig.app_title || filename || "Untitled Notebook";
-  }, [appConfig.app_title, filename]);
+    document.title = appConfig.app_title || filetitle || "Untitled Notebook";
```
Comment (mscolnick) [inline frontend/src/core/edit-app.tsx:136]: `appConfig.app_title || Paths.basename(filename) || "Untitled Notebook";`

File: frontend/src/core/edit-app.tsx (hunk 2 above)
Comment (ericjanto) [reply frontend/src/core/edit-app.tsx:124]: Got it–ignore my recent commit.

**Review by mscolnick — APPROVED** [review]

**Pull request merged**